protocol/communities/manager.go

```diff
@@ -81,7 +83,19 @@ func (t *HistoryArchiveDownloadTask) Cancel() {
 	close(t.CancelChan)
 }
 
-func NewManager(identity *ecdsa.PrivateKey, db *sql.DB, encryptor *encryption.Protocol, logger *zap.Logger, verifier *ens.Verifier, transport *transport.Transport, torrentConfig *params.TorrentConfig) (*Manager, error) {
+type managerOptions struct {
+	accountsManager *account.GethManager
+}
+
+type ManagerOption func(*managerOptions)
+
+func WithAccountManager(accountsManager *account.GethManager) ManagerOption {
+	return func(opts *managerOptions) {
+		opts.accountsManager = accountsManager
+	}
+}
```

Comment:
@Samyoul FYI introduced `managerOptions` here. This will extend in the follow-up PRs